src/daemon.js

```diff
@@ -0,0 +1,98 @@
+const { info } = require('./log');
+
+const LEGACY_0_8_UPGRADE_DOC = '_design/medic:staged';
+const HORTI_UPGRADE_DOC = 'horti-upgrade';
+
+const Apps = require('./apps'),
+      DB = require('./dbs'),
+      install = require('./install'),
+      fatality = require('./fatality');
+
+const newDeployment = deployDoc =>
+  !!deployDoc &&
+  deployDoc._id === HORTI_UPGRADE_DOC &&
+  (deployDoc.action !== 'stage' || !deployDoc.staging_complete);
```

Comment:
I think pull the various state strings out as constants. It'd be a good way to document what each stage means and display all possible stages in one place.

Reply:
Yeah good idea. I also copy constants around in places (the horror!). I will put them all in one place